EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
 
-        private GuildDialogState _state;
+        private readonly Stack<State> _stateStack;
+        private int _creationMemberCount;
 
+        private State _state;
         public GuildDialog(INativeGraphicsManager nativeGraphicsManager,
                          IEODialogButtonService dialogButtonService,
                          IEODialogIconService dialogIconService,
                          ILocalizedStringFinder localizedStringFinder,
                          ICharacterProvider characterProvider,
-                         IEOMessageBoxFactory messageBoxFactory)
+                         IEOMessageBoxFactory messageBoxFactory,
+                         ITextMultiInputDialogFactory textMultiInputDialogFactory,
+                         IContentProvider contentProvider,
+                         IGuildActions guildActions,
+                         IGuildSessionProvider guildSessionProvider,
+                         ICharacterInventoryProvider characterInventoryProvider,
+                         IEIFFileProvider eifFileProvider)
             : base(nativeGraphicsManager, dialogButtonService, DialogType.Guild)
         {
             _dialogIconService = dialogIconService;
             _localizedStringFinder = localizedStringFinder;
             _characterProvider = characterProvider;
             _messageBoxFactory = messageBoxFactory;
+            _textMultiInputDialogFactory = textMultiInputDialogFactory;
+            _contentProvider = contentProvider;
+            _guildActions = guildActions;
+            _guildSessionProvider = guildSessionProvider;
+            _characterInventoryProvider = characterInventoryProvider;
+            _eifFileProvider = eifFileProvider;
+            _stateStack = new Stack<State>();
+            _creationMemberCount = 0;
 
-            SetState(GuildDialogState.Initial);
+            SetState(new State(GuildDialogState.Initial));
 
             BackAction += (_, _) =>
             {
-                switch (_state)
+                if (_stateStack.Count > 0)
                 {
-                    case GuildDialogState.Management:
-                        SetState(GuildDialogState.Initial);
-                        break;
-                    default:
-                        // no-op
-                        break;
+                    var previousState = _stateStack.Pop();
+                    SetState(previousState);
+                }
+                else
+                {
+                    SetState(new State(GuildDialogState.Initial));
                 }
             };
 
             Title = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_MASTER);
         }
 
-        private void SetState(GuildDialogState state)
+        protected override void OnUpdateControl(GameTime gameTime)
         {
-            if (state != GuildDialogState.Initial && _state == state)
-                return;
+            if (_state.DialogState == GuildDialogState.RegisterGuild)
```

Comment:
I'm slightly concerned about the approach to poll for `Approved` in this way...but I don't know if I can think of a different way to make it work.